remove duplicated observations

IJBG opened the pull request.

IJBG commented on R/mbbs.R:
```diff
@@ -36,24 +36,38 @@ create_stop_level_counts_0 <- function(ebird, taxonomy, config = config) {
               "in {x$year} coming from {x$sources}"
             )
           )
         })()
+      
+      # Log number of duplicate entries with the same count
+      df |>
+        dplyr::distinct(year, route, stop_num, common_name, count,
+                        .keep_all = TRUE) |>
+        (\(x) {
+          logger::log_trace(
+            paste("Stop-level data contains",
+                  "{nrow(df) - nrow(x)} duplicated observations",
+                  "(same count, different sources)"))
+        })()
```
I think I'm following the same logic here as elsewhere in the script with the loggers, but not getting anything to print out, not 100% clear to me how the df continues to be passed along within the validity checks, but adding a pipe to line 58 threw an error `Error in df : 
  The pipe operator requires a function call as RHS (<input>:21:7)`

IJBG commented on R/mbbs.R:
```diff
@@ -37,23 +37,41 @@ create_stop_level_counts_0 <- function(ebird, taxonomy, config = config) {
               "in {x$year} coming from {x$sources}"
             )
           )
         })()
+
+      # Log number of duplicate entries with the same count
+      df |>
+        dplyr::distinct(year, route, stop_num, common_name, count,
+          .keep_all = TRUE
+        ) |>
+        (\(x) {
+          logger::log_trace(
+            paste(
+              "Stop-level data contains",
+              "{nrow(df) - nrow(x)} duplicated observations",
+              "(same count, different sources)"
+            )
+          )
+        })()
+
```
I think I'm following the same logic here as elsewhere in the script with the loggers, but not getting anything to print out, not 100% clear to me how the df continues to be passed along within the validity checks, but adding a pipe to line 58 threw an error Error in df :  The pipe operator requires a function call as RHS (<input>:21:7)

IJBG replied:
Ah! Okay, it's not logging because it's only a log_trace, and the 'level' is too low to print. Working as intended.
"Stop-level data contains 7889 duplicated observations (same count, different sources)"

bsaul replied:
Indeed you can set the log level with the log threshold function: https://daroczig.github.io/logger/reference/log_threshold.html

IJBG commented:
Went and fixed some locations and checklists with problems in ebird (>20 records from obs notes, and NAs for stop_num 1) we'll be due for another download from ebird soon. Now 306 warnings for mismatches, and with the easy ebird fixes I did probabllyyyy 250, down from 691

bsaul commented:
> Went and fixed some locations and checklists with problems in ebird (>20 records from obs notes, and NAs for stop_num 1) we'll be due for another download from ebird soon. Now 306 warnings for mismatches, and with the easy ebird fixes I did probabllyyyy 250, down from (the incorrect estimate w duplicates) 691

Cool!

IJBG commented on R/mbbs.R:
```diff
@@ -107,12 +133,45 @@ create_stop_level_counts <- function(ebird, taxonomy, config = config) {
+          paste(
+            "{x$year}-{x$route} had",
+            "{x$scount} {x$common_name} aggregrated in stop_level",
+            "{x$source}",
```
Adding 'source' here and lines 135 are the changes in this section from the other pull request

IJBG commented:
Okay! Removed the observer quality calculations and shifted things over to the config$ format, but there's still some reworking to be done for sure to work with the new formatting, eg. no separation of processing by county anymore. Added a couple logger:: calls where I'd had prints() I used in testing commented out.

I removed propogate_observers_across_stops because we don't need it anymore, observer is only going into the survey_events files anyway

I also replaced some calls with config$ in survey-events while I was at it

bsaul commented:
Looks good to me.

bsaul commented:
@IJBG - is this ready to merge? I'm going to wait on this to merge b/f going into observers/etc.

IJBG commented:
Ready to merge!

IJBG commented:
Just needs the minor conflict fixed if you want to resolve that - I don't have a laptop on me right now but will in an hour, just wrapped up judging the Chatham science fair ⚗️
